### ISSUE TITLE
How to output the fal(Forecast albedo) by using ece2cmor3?

### ISSUE THREAD

**Issue by hywzq** [EC-Earth/ece2cmor3#774]
Hi,
I am writing to seek your assistance with an issue I am facing while using ece2cmor3 to process EC-EARTH3 output data.
When using ece2cmor3 to process the original data, I have observed that it does not include the 'fal' variable in the output data. However, the 'fal' variable is present in the original data. 
I have attempted to adjust the 'CMIP6_Amon.json, cmip6-data-request-varlist-PMIP-transientHolocene-EC-EARTH-Veg-LR.json' but have not been successful in resolving the issue.
Could you give me any ideas?

Best,
Zhenqian

**Comment by klauswyser** [comment]
There is no "albedo" in the CMIP6 data request and therefore no method in ece2cmor3 to save FAL in a CMIP6 compliant format.

You could tweak the system: add a new "albedo" variable to the CMIP6 tables, add a "recipe" that links FAL and your new variable in ifspar.json, and then add the variable to your varlist.json. There is a more _official_ way to do this, add_extra_output_variables (link?) that @treerink could explain.

**Comment by treerink** [comment]
Hi Zhenqian, I sent you a mail (I hope I found the correct address), so we can schedule a short online meeting.

**Comment by hywzq** [comment]
> There is no "albedo" in the CMIP6 data request and therefore no method in ece2cmor3 to save FAL in a CMIP6 compliant format.
> 
> You could tweak the system: add a new "albedo" variable to the CMIP6 tables, add a "recipe" that links FAL and your new variable in ifspar.json, and then add the variable to your varlist.json. There is a more _official_ way to do this, add_extra_output_variables (link?) that @treerink could explain.

Thank you @klauswyser.
Sorry for my late response.  I will contact Thomas for assistance.

**Comment by treerink** [comment]
Ok thus this is what we want to add: monthly output:
`Amon`  `fal`
with grib code `243.128`

**Comment by treerink** [comment]
project: multi centennial climate variability  at Stockholm University SU-request

The request is similar to the monthly-only  request of:
`ctrl/output-control-files/cmip6/CMIP/EC-EARTH-Veg-LR/cmip6-experiment-CMIP-piControl`

The idea is that this `Amon fal` shows an example, with which more variables can be added.

**Comment by treerink** [comment]
Hi Zhenqian, 

After a `git pull` you now can run from the `ece2cmor/ece2cmor/scripts` dir:

`./genecec-for-individual-experiments.sh ../resources/miscellaneous-data-requests/su-climvar/varlist-su-multi-centennial-climate-variability.json CMIP piControl  EC-EARTH-Veg-LR su-multi-centennial-climvar`

This would do the preparation and the `add-su-climvar-variables.sh` script shows the format for further extension. But first it would be good to test your cmorisation with this for `Amon fal`. The produced `su-multi-centennial-climvar/` directory is the one you have to specify in your ECE `config-run.xml` in the var `OUTPUT_CONTROL_FILES_DIR`.

Let me know.

**Comment by hywzq** [comment]
Hi Thomas,
Thanks for your help.
I have successfully add the 'fal' and run cmor in our HPC.
Now, I am trying to follow your methods to add other variables in cmor.
Thanks again.

Best,
Zhenqian

**Comment by hywzq** [comment]
Hi Thomas (@treerink ),
I hope this email finds you well. I have a few questions that require your expertise. Specifically,  when the start year of my simulation experiment is set to 1000 or earlier than 1850, CMOR encounters an error, as shown below. 
LPJG:
```
2023-11-27 19:31:59 ERROR:ece2cmor3.lpjg2cmor: The reference date given is after the first year in the data (1000) for variable cSoil in file /ec/res4/scratch/src/ecearth3/output/tmp/001_LMTT_lpj/cSoil_monthly.out. Skipping CMORization.
```
NEMO:
```
! Error: axis time (table: Omon) has non monotonic values:
! -17973.500000, -17943.000000 and  -18095.000000
```

I believe the issue is related to the **reference date**, but I haven't found the specific command defining the reference date. Additionally, I noticed in **nemo2cmor.py** and **ifs2cmor.py**, there are lines and comments as follows:
```
timeshift = timedelta(0)
# Apply timeshift for instance in case you want manually to add a shift for the piControl:
# timeshift = datetime(2260, 1, 1) - datetime(1850, 1, 1)
```
I don't understand the role of this timeshift in CMOR. 
Could you give some insights into these two issues? 

Thank you!

Best,
Zhenqian

**Comment by treerink** [comment]
Can you try to set the reference date for output time axes (default: 1850-01-01)
 `--refd YYYY-mm-dd `
as option to ece2cmor3, this:
 `--refd 1000-01-01 `

**Comment by hywzq** [comment]
> Can you try to set the reference date for output time axes (default: 1850-01-01) `--refd YYYY-mm-dd ` as option to ece2cmor3, this: `--refd 1000-01-01 `

Thank you for your prompt response.
Could you please provide more details on where exactly I should apply this setting?

**Comment by goord** [comment]
Hi @hywzq it's an extra argument to the ec2cmor command

**Comment by hywzq** [comment]
> Hi @hywzq it's an extra argument to the ec2cmor command

Thank you for the clarification. I will proceed to apply this option and see if it resolves the issue.
By the way, do you have any additional insights or recommendations related to the 'timeshift'？
Thanks again for your help.

**Comment by treerink** [comment]
As you see the:
`# timeshift = datetime(2260, 1, 1) - datetime(1850, 1, 1)`
is in comment. It has been used (a tiny hack) to shiftback the final state of a certain piControl run after its long spin-up in order to match its piControl 1850 intended date. Just to have 1850 as date in the piControl run from which the historical runs can start.

**Comment by treerink** [comment]
The other thing you need to keep an eye on is the `"parent_time_units"` in the metadata template files, for instance in the fresh created `su-multi-centennial-climvar/metadata-cmip6-CMIP-piControl-EC-EARTH-Veg-LR-lpjg-template.json` this is now:
`    "parent_time_units":            "days since 1850-01-01",`

**Comment by treerink** [comment]
You can just adjust this date such that it fits your case.

**Comment by treerink** [comment]
For LPJG, yes indeed there we have for the current python3 version bug #771. As you confirm, giving only zero's. So yes I am aware, this ug needs to be solved.

**Comment by treerink** [comment]
For your NEMO output, are you sure this is unmodified original ECE raw output which you are cmorising, because it says `non monotonic values` and indeed the last printed time step is earlier than the one before?

**Comment by hywzq** [comment]
> For your NEMO output, are you sure this is unmodified original ECE raw output which you are cmorising, because it says `non monotonic values` and indeed the last printed time step is earlier than the one before?

I am sure, those data is unmodified original ECE raw output.

**Comment by treerink** [comment]
Because the original request has been implemented and meanwhile #771 is solved, I am closing this issue. Please open a new one when experiencing difficulties (eventually with a reference to this issue).
